server/src/main/java/com/cloud/network/router/NetworkHelperImpl.java

```diff
@@ -551,6 +567,10 @@ public DomainRouterVO deployRouter(final RouterDeploymentDefinition routerDeploy
                     s_logger.debug("Failed to allocate the VR with hypervisor type " + hType + ", retrying one more time");
                     continue;
                 } else {
+                    // If VR can't be deployed then decrement the resource count
+                    if (Boolean.TRUE.equals(VirtualMachineManager.ResourceCountRouters.valueIn(owner.getDomainId()))) {
+                        _itMgr.decrementVrResourceCount(routerOffering, owner,true);
+                    }
```

Comment:
sounds like a utility method and the comment as javadoc. Can you extracta and reuse this, please?